templates/show_entries.html

```diff
@@ -14,6 +14,8 @@
   <ul class=entries>
   {% for entry in entries %}
     <li><h2>{{ entry.title }}</h2>{{ entry.text|safe }}
+    <a href="/entry={{ entry.id }}&move=up">Up</a>
```

Comment:
Yes, it is better to have them as POSTs.

Two solutions here: use the form with button (you can still style it as a link) or simple jQuery script to make this link perform the POST request (the link will have no "href" and the actual work will be done in the onclick event handler).